core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -43,7 +43,8 @@
 cudaStream_t Kokkos::Impl::cuda_get_deep_copy_stream() {
   static cudaStream_t s = nullptr;
   if (s == nullptr) {
-    cudaStreamCreate(&s);
+    KOKKOS_IMPL_CUDA_SAFE_CALL(
+        (CudaInternal::singleton().cuda_stream_create_wrapper(&s)));
```

Comment:
based on the discussion of Wed 26 july, can we please open an issue where we can the name of this `singleton` into something else since it is not a singleton?